primal-bit/src/lib.rs

```diff
@@ -171,11 +170,21 @@ impl BitVec {
     /// }
     /// ```
     pub fn from_elem(nbits: usize, bit: bool) -> BitVec {
+        use std::ptr;
+
         let nblocks = nbits.checked_add(BITS - 1).expect("capacity overflow") / BITS;
+        let mut out_vec = Vec::with_capacity(nblocks);
+
+        unsafe {
+            out_vec.set_len(nblocks);
+            ptr::write_bytes(out_vec.as_mut_ptr(), if bit { !0 } else { 0 }, nblocks);
+        }
```

Comment:
The standard library uses unstable features **a lot**, and in [macros like `vec!`](https://doc.rust-lang.org/nightly/std/ops/trait.Try.html) it uses `#[allow_internal_unstable]`, so it can even use unstable features when expanded in your stable code.  This case will use [`vec::from_elem`](https://github.com/rust-lang/rust/blob/b8499e0460e36dfb304b8800c83587b2cacc887c/src/liballoc/vec.rs#L1445-L1447) which indeed uses specialization.

Given that it's able to do such tricks, let's at least *try* `vec!` to compare performance, OK?